go/framework/bertybridge/bridge.go

```diff
@@ -65,15 +77,63 @@ func NewBridge(config *Config) (*Bridge, error) {
 	}
 
 	// setup logger
-	{
-		if nativeLogger := config.dLogger; nativeLogger != nil {
-			b.logger = newLogger(nativeLogger)
-		} else {
-			b.logger = zap.NewNop()
+
+	cores := []zapcore.Core(nil)
+
+	// tyber core
+	if len(config.TyberHost) != 0 {
+		mutex := &sync.Mutex{}
+		canceled := false
+		ch := make(chan struct{})
+		go func() {
+			defer func() { ch <- struct{}{}; close(ch) }()
+			if logger, clean, err := logutil.NewLogger(logutil.NewTyberStream(config.TyberHost)); err == nil {
+				mutex.Lock()
+				defer mutex.Unlock()
+				if canceled {
+					clean()
+					return
+				}
+				b.closeLogger = clean
+				cores = append(cores, logger.Core())
+			}
+		}()
+		select {
+		case <-time.After(time.Second * 2):
+			mutex.Lock()
+			canceled = true
+			mutex.Unlock()
+		case <-ch:
 		}
+	}
+
+	// native core
+	if nativeLogger := config.dLogger; nativeLogger != nil {
+		nl := newLogger(nativeLogger)
+		cores = append(cores, nl.Core())
+	}
+
+	// create logger
+	if len(cores) == 0 {
+		b.logger = zap.NewNop()
+	} else {
+		b.logger = zap.New(
+			zapcore.NewTee(cores...),
+			zap.AddCaller(),
+		)
+		b.logger.Debug("logger initialized", zap.Any("manager", &initutil.Manager{
+			SessionID: randomID(),
```

Comment:
this is super hacky, you should just let initutil.Manager makes the stuff by itself

what should be done:
* update `bertyaccount/service_account.go` to use `initutil.New()` instead of `initutil.Manager{}`, so it creates an uuid session.ID
* call `manager.SetLogger(b.logger)`

and at the end, you can do this
```suggestion
		b.logger.Debug("logger initialized", zap.Any("manager", manager))
```

Reply:
if my PR is merged, I suggest you to just rebase master and apply the last changes -> https://github.com/berty/berty/pull/3334

Reply:
this happens before the manager gets initialized
I think it's good that the logger gets initialized first so we get nformation even in case of early errors
I think it would be better to define a format for the tyber header instead of logging the manager struct

Reply:
> I think it's good that the logger gets initialized first so we get nformation even in case of early errors

No, it's clearly not worth it, early error should simply prevent the app to start, it's far more confusing to have this here, especially without any comment around 

> I think it would be better to define a format for the tyber header instead of logging the manager struct

we should definitely avoid the use of `Any` and only log useful field of the manager (if you really want to log this)

Reply:
> we should definitely avoid the use of Any and only log useful field of the manager (if you really want to log this)
it's they way tyber was set to be initialized, I agree we should change it but I deemed it out of scope of this PR

Reply:
From my side I think it's good enough as it is at the moment. I used [this log](https://github.com/berty/berty/blob/1cc215104c4ec34138fffb2229f79573a6225a4f/go/internal/initutil/logger.go#L85) for the Tyber init (which already existed) when I made the v1.
It would just be necessary as @moul says that the mobile also goes through this for its init to unify, rather than dupplying this part with a "hack".